Adaptive Importance Sampler class

somu15 opened the pull request:
## Reason
Adaptive Importance Sampler is used for estimating small failure probabilities efficiently using expensive FE models.

## Design
Three classes have been implemented:

- AIS.C sampler class for proposing importance samples
- AdaptiveMonteCarloDecision.C reporter class for generic decision-making support for adaptive Monte Carlo sampling using the present implementation and planned future implementations.
- AdaptiveMonteCarloUtils.C utils class which will contain functionalities required by the present implementation and planned future implementations of adaptive Monte Carlo samplers.

## Impact
No impact on existing objects.

Closes #17664

lindsayad committed:
Create fluid-structure example. Refs #12462

andrsd committed:
Adding enthalpy injection BC

Refs #15742

andrsd committed:
Adding action to set up enthalpy injection BC

Currently we allow only pressure and temperature variables for this BC.

Refs #15742

andrsd committed:
Adding jacobian test

Refs #15742

fdkong committed:
Create fluid structure interaction module

Closes #12462

somu15 committed:
FSI codes #15712

aeslaughter committed:
Add SQA report files

(refs #15995)

somu15 committed:
Update config.yml

aeslaughter committed:
Remove simple diffusion test

(refs #15995)

aeslaughter committed:
Complete SQA setup

(refs #15995)

somu15 committed:
Update AcousticInertia.h

somu15 committed:
code cleanup

fdkong committed:
Rename fluid_structure_interaction to fsi

since fluid_structure_interaction is too long

somu15 committed:
Update index.md

fdkong committed:
Disable ASSERT_NO_THROW(1 + 1) in unit test

somu15 committed:
Update FluidStructureInterface.C

somu15 committed:
Update FluidStructureInterface.C

GiudGiud committed:
Add convective heat transfer term, refs #16756

GiudGiud committed:
Add energy time derivatice porous flow kernel

GiudGiud committed:
Add PINSFV Energy advection kernels

GiudGiud committed:
Add porous flow INSFV energy diffusion kernel

GiudGiud committed:
Add porous media INSFV mass equation kernels

GiudGiud committed:
Add porous media momentum time derivative, simplify some other terms through inheritance

GiudGiud committed:
Add PINSFVVelocityVariable to force superficial velocity type
Add momentum advection kernels

GiudGiud committed:
Add momentum diffusion with spatially constant porosity

GiudGiud committed:
Add porosity gravity and boussineq kernels

GiudGiud committed:
Rename gravity and boussinesq kernels for their respective variable
Add momentum friction kernel

GiudGiud committed:
Add momentum pressure gradient term kernel, refs #16756

GiudGiud committed:
Add a basic channel test

GiudGiud committed:
Got the porosity face value approach for momentum advection

GiudGiud committed:
Add documentation to FV Dirichlet boundary conditions

dewenyushu committed:
Add ComputeMultipleCrystalPlasticityStress and CrystalPlasticityStressUpdateBase classes

Remove original CP stress update version

Transform Kalidindi to ComputeMultipleCrystalPlasticityStress-based while retaining its original performance

Update documentation

Refs issue #16064

GiudGiud committed:
Fix momentum diffusion term, only enable porosity gradient term for smooth porosity
Interpolate k_eps on face for energy diffusion

GiudGiud committed:
Remove DOFmap printing from INSFV tests

GiudGiud committed:
Add ifdef against local AD indexing, first stab at pinsfv tests

GiudGiud committed:
Move files to match conventional moose file locations

GiudGiud committed:
Rename Gravity and Boussinesq force kernels to highlight that they apply to the momentum equation

GiudGiud committed:
Interpolate porosity with the other terms for momentum advection

GiudGiud committed:
Add symmetry BCs, rename heat convection, add heated test

GiudGiud committed:
Add PINSFV conservation test

GiudGiud committed:
Add porosity to Rhie Chow interpolation, pressure and diffusion terms

GiudGiud committed:
Add more options to PINSFV friction models

GiudGiud committed:
Remove porosity derivative term

GiudGiud committed:
Fix some porous kernels

GiudGiud committed:
Add effective thermal diffusion kernel

GiudGiud committed:
Improve params documentation
Compute porosity on the face for symmetry BCs

GiudGiud committed:
Add stubs

GiudGiud committed:
Add discontinuous porosity test

GiudGiud committed:
Add a flux version of the momentum equation pressure term

GiudGiud committed:
Add a porosity discontinuity term which causes a proportional pressure jump

GiudGiud committed:
Add a Generic vector material for testing anisotropic friction kernels

aprilnovak committed:
Add member function to determine target nodes.

aprilnovak committed:
Add member function to determine target elems.

GiudGiud committed:
Add 1D porosity jump test, clean up 2D test
Add porosity jump treatment to advection kernel
Add continuous porosity gradient term from advection kernel
Add effective viscosity option to momentum diffusion
Add momentum matprop option to friction kernel, switch to momentum for consistency with Pronghorn
Fix formulation of porosity gradient term

GiudGiud committed:
Update documentation
Express friction in terms of superficial velocity
Remove mass time derivative kernel
Remove option for effective viscosity in momentum diffusion kernel

GiudGiud committed:
Clang formatting
Fix eps bug in friction

GiudGiud committed:
Clang formatting

GiudGiud committed:
Documentation fixup, add PINSFV main documentation file, refs #16756

aprilnovak committed:
Add test for target_boundary for CONSTANT monomial. Refs #17214

GiudGiud committed:
Switch PINSFVVelocityVariable to SuperficialVelocityVariable

aprilnovak committed:
Add member function for finding the maximum nearest distance to avoid redundancy. Refs #17214

aeslaughter committed:
Mar. 2021 news

(refs #11447)

aeslaughter committed:
Mar. 2021 news

(refs #11447)

aprilnovak committed:
Add error to prevent usage of first or higher order monomials with nearest node transfer. Refs #17246

GiudGiud committed:
Temporary fix on FVBCs for neighbor element evaluations
Fix PINSFVSymmetryVelocityBC issue with porosity value
Clean up of MomentumPressureFlux
Make energy advection use porous RC
Tolerance on floating point math for disabling RC at porosity jumps

YaqiWang committed:
add paragraph for early construction of subapps

somu15 committed:
FSI interface update

somu15 committed:
Fix conflicts

GiudGiud committed:
Add check on porous variable type for porous kernels

roystgnr committed:
Include some of the libMesh update changes

fdkong committed:
Moved Alex's 2D example to the right place

fdkong committed:
2D toy example works

Closes #12462

recuero committed:
Promote the computation of strain energy rate density to a parent class.

Now that StressUpdateBase handles the computation of strain energy rate density,
the functionality that relies on this computation can work on constitutive laws
that do not inherit from our RadialReturn classes.

Refs. #15517

GiudGiud committed:
Fix some PINSFV documentation
Fix GenericVectorMaterial for AD

GiudGiud committed:
Clean up and update PINSFV tests, add friction and boussinesq tests, refs #16755 #16872

GiudGiud committed:
Experimental treatment of porosity jumps in momentum pressure term

aeslaughter committed:
Start Apr. news

(refs #11447)

GiudGiud committed:
Improve headers
Allow user to specify what consistutes a discontinuity in porosity

GiudGiud committed:
Add all test gold files for PINSFV
Add some variable type checks
Clean source files, leave TODOs for namespace issue with superficial velocity in continuous porosity case
Add a non-unity porosity in conservation test
Update momentum pressure flux kernel to evaluate eps*P on faces
Use PINSFV outflow kernels instead of INSFV for mean pressure channel
Use LU preconditioner for tests that fail in parallel
Remove triangle mesh test for now

Address Alex's 2 reviews
Fix BC execution on neighbor side
Fix ConstantVectorMaterials
Fix documentation
Re-use rcCoeff from INSFVMomentumAdvection

Fix parallel bug in PINSFV momentum advection
Fix potential for bad material properties in MomentumDiffusion
Add 1D and 3D tests to INSFV

Remove linear and quadratic generic friction model, which are present in INSFV

Re-gold tests
Add test for effective diffusion of energy

Use non-unity porosity in heated tests, re-gold

Fix for sparse AD indexing compile
Clang format

fdkong committed:
Fande's contributions

elementx54 committed:
Add Parameter to Enable Final Jacobian Testing

The new parameter in `PetscJacobianTester.py` allows for testing of only
the final Jacobian comparison. The default behavior is only changed when
the new parameter is used.

Added a tensor mechanics test using the new parameter. If the new
parameter is removed, this new test would fail due to other Jacobian
comparisons being above the tolerance ratio.

Closes #17456

fdkong committed:
Moved MatReaction.{C,h,md} to the framework from phase_field module

fdkong committed:
Moved INSConvectedMesh.{C,h,md} to fsi since it is part of ALE

Renamed it to ConvectedMesh.{C,h,md} since it should for both compressible and incompressible flows

fdkong committed:
Moved CoefReaction.{C,h,md} to the framework

and deleted CoefReaction.{C,h} from MooseTestAPP.

We should not duplicate the code

aeslaughter committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

lindsayad committed:
Alex additions

dschwen committed:
Fix typo (#17683)

loganharbour committed:
Improve PostprocessorInterface

- Make const-correct where possible
- Use the related object's paramError and mooseError where possible
- Make Reporter more const-correct without the need for a write key
  for the PostprocessorInterface
- Improve the checking of the existance of a postprocessor
- Remove metadata store for postprocessors in InputParameters/Parser
  and instead check on the fly

Refs #17512

loganharbour committed:
Move UserObject methods to source/bottom of header

refs #17512

loganharbour committed:
Make interface methods that are overridden virtual

refs #17512

loganharbour committed:
Use UserObjectInterface methods with error checking

refs #17512

loganharbour committed:
Add tests for PostprocessorInterface errors

refs #17512

loganharbour committed:
Add object information to paramError when the file location is not available

loganharbour committed:
Make sure function signatures are consistent

refs #17512

loganharbour committed:
Improve VectorPostprocessorInterface

- Use the object's mooseError() and paramError() where possible
- Improve function signatures
- Use error checking in overridden methods

refs #17512

loganharbour committed:
Check that in the case of non-insertion that values are the same

refs #17512

loganharbour committed:
Remove unnecessary reporter getter in FEProblemBase

refs #17512

aeslaughter committed:
Update PR template to align with coming changes to SQA process

(refs #17540)

aeslaughter committed:
Update PR template to align with coming changes to SQA process

(refs #17540)

vanwdani committed:
Additional refinement
Refs #14823

vanwdani committed:
Additional refinement
Refs #14823

aeslaughter committed:
Update .github/PULL_REQUEST_TEMPLATE.md

Co-authored-by: Daniel Schwen <daniel@schwen.de>

hugary1995 committed:
Block-restricted DiracKernel

closes #17561

GiudGiud committed:
Add documentation for the integral variable pp, refs #17566

loganharbour committed:
Remove unneeded hasPostprocesorObject method

refs #17512

YaqiWang committed:
add tag gradient #17586

YaqiWang committed:
break header dependency in material property interface #17586

YaqiWang committed:
add missing pieces for coupling tagged solution vectors #17586

loganharbour committed:
Add extra error checking when possible, add enabled pp method

refs #17512

aeslaughter committed:
Update default priority and language

(refs #17540)

YaqiWang committed:
add more interface functions in Coupleable #17586

YaqiWang committed:
fix constant monomial variable for coupling tagged solution vectors #17586

YaqiWang committed:
add tests for tagged coupling #17586

dschwen committed:
Fix winding order (#17683)

loganharbour committed:
Add dependency helpers to PP, VPP, and UO interfaces

refs #17512

loganharbour committed:
Improve UserObjectInterface error testing

refs #17512

loganharbour committed:
Various fixups

refs #17512

dschwen committed:
Add winding order test aux kernel (#17683)

loganharbour committed:
Add more tests for Reporter and ReporterInterface

refs #17512 #17605

loganharbour committed:
Reporter system enhancements

- Declare Reporter values for PPs and VPPs within the objects themselves
- Add type checking on declare/request for Reporters with the same name
- Add producer/consumer MooseObject ptrs in ReporterState
- Add getter for information about a Reporter (type, producers, consumers,
  context, modes)
- Add more error handling to ReporterData, Reporter, and ReporterInterface

refs #17512 #17605

GiudGiud committed:
Fix nomenclature, refs #15457

loganharbour committed:
Add object for outputting Reporter debug information

refs #17605

aeslaughter committed:
Changes for TestHarness migration to moosetools

(refs #17006)

zachmprince committed:
Using parameter to repartition sampler #15787

hugary1995 committed:
address some comments

zachmprince committed:
refactoring out rank/app partitioning algorithms with nice unit tests #15787

Co-authored-by: rwcarlsen <rwcarlsen@gmail.com>

aprilnovak committed:
Add normal component to DiffusionFluxAux. Refs #17625

joshuahansel committed:
Allowed 2-phase interfaces to be called in TwoPhaseFluidPropertiesIndependent

Closes #17632

YaqiWang committed:
a fix for time integrator: associate vectors with tags early because kernels may be using the tags #17586

loganharbour committed:
Make clang-format happy

loganharbour committed:
Add test design/doco pages for Reporter and VPP interfaces

refs #17605 #17512

loganharbour committed:
Re-add unused initPostprocessorData to be removed soon

refs #17512

loganharbour committed:
Move over-zealous assertion

refs #17605

loganharbour committed:
Clean up PP interface for default values and remove deprecated methods

refs #17512 #17605

loganharbour committed:
Re-add deprecated param that isn't ready to go yet

refs #17512

joshuahansel committed:
Implemented derivatives overload of IdealGasFluidProperties::k_from_v_e

Refs #12815

WilkAndy committed:
CappedMohrCoulombStressUpdate includes corner case

Fixes #17648

dschwen committed:
Allow for multiple timestep limiting functions (#17642)

somu15 committed:
Addressing comments 1

somu15 committed:
Addressing comments 2

GiudGiud committed:
Fix up documentation for variablegradientcomponent, refs #17566

YaqiWang committed:
add tags for normal solution vectors (requires users to indicate solution vector of previous Newton iteration is required or not) #17586

loganharbour committed:
Remove unused method

refs #17512

loganharbour committed:
Move Reporter producer ownership to the context

refs #17605

somu15 committed:
AIS sampler class

somu15 committed:
Adaptive Monte Carlo Decision class

somu15 committed:
Adaptive Monte Carlo Utils class

somu15 committed:
Create tests

YaqiWang committed:
enable field transfers with tagged solution vector for #17586

somu15 committed:
Create docs #17664

somu15 committed:
Create docs #17664

dschwen committed:
Add boundary marker (#17661)

aeslaughter committed:
Fix comments in template

(refs #17540)

YaqiWang committed:
fix array variable relaxation in Picard iteration #17659

dschwen committed:
Distance (#17661)

hugary1995 committed:
Point-not-found behavior

loganharbour committed:
Revert "Allow target_boundary for elemental nearest point transfers"

aeslaughter committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

YaqiWang committed:
support tagging the current solution #17586

YaqiWang committed:
we need for now to shortcut a few calls in MooseVariableData for special tags otherwise end up with redundant evaluations #17586

dschwen committed:
Add test and docs (#17661)

fdkong committed:
Index into Assembly local vectors using _type_id since TaggingInterface
and Assembly handle only residual vectors

recuero committed:
Avoid throwing exception when generated mortar segments have a dubious normal. Instead, do not perform the splitting procedure
as this case has shown (so far) to correspond to extremely oblique projections.

Closes #17696

joshuahansel committed:
Added PostprocessorInterface to InitialConditionBase

- Added PostprocessorInterface to InitialConditionBase
- Added overloads for InitialConditionBase::getPostprocessorValue
  and InitialConditionBase::getPostprocessorValueByName that add
  the post-processor name to the user object dependency list
- Tested correct dependency resolution of both new overloads
- Added news entry

Closes #17692

Commit:
Implement const monomial support for MultiAppPostprocessorInterpolationTransfer and check to_var fe type (#14631)

loganharbour committed:
Add preTrace() and always call postOnSegment()

refs #17702 and #17701

loganharbour committed:
Simplify ClaimRays()

refs #17703

dschwen committed:
Abaqus UMAT (#14974)

recuero committed:
Improve description of mortar's invalid orientation message

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

dschwen committed:
Add elastic test and redo old hardening test (#14974)

dschwen committed:
Recover plugins (#14974)

hugary1995 committed:
Address comments, add another block-restriction check

YaqiWang committed:
fix a memory issue revealed by a test in #17672

dschwen committed:
Address comments (#17661)

aeslaughter committed:
Remove requirement traceability links from SRS template

(refs #17540)

GiudGiud committed:
Apply suggestions from code review

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

GiudGiud committed:
Re-organize PINSFV channel tests

GiudGiud committed:
Add documentation to generic materials, refs #17566

loganharbour committed:
Do not report root Reporter values on non-root procs

refs #17712

dschwen committed:
Pass through jacobian, simplify (#14974)

permcody committed:
Fix typo introduced in previous commit

refs #14767

aeslaughter committed:
Create Communication and Contact Information documents for SQA

(refs #17540)

aeslaughter committed:
Support multiple directories when searching for files in SQA reports

(refs #17540)

aeslaughter committed:
Update contributing and documentation pages

(refs #17540)

lindsayad committed:
Use one-term boundary expansion by default for INSFV

This fixes #16822 although perhaps not 100% satisfactorily. If a user
knows they only have cells with a maximum of one extrapolated boundary
face or if they know that if the cell has multiple extrapolated boundary
faces that it is orthogonal (e.g. dCf is parallel to Sf), then they can
safely set `two_term_boundary_expansion = true` in their input. But in
the interest of allowing more things to just work out of the box (like
any old tri or tet mesh), I think it makes sense to make the one term
expansion the default. I prefer doing this over something like
projecting dCf onto the surface normal because this would cost a user
accuracy in the case where they have intelligently designed their mesh
such that there is only a maximum of one extrapolated boundary face per
cell (gmsh meshes for example avoid this).

aeslaughter committed:
Add code standards to list of SQA documents

(refs #17540)

GiudGiud committed:
Fix up documentation
Address Alex's review:
- add docstring to fvbcs
- fix equation in momentum outflow bc
- use solid phase instead of local medium in test docs
- break up pinsfv channel test directory

aeslaughter committed:
Fix acronym spelling mistakes

(refs #17540)

aeslaughter committed:
Add Software Library List to SQA documentation and shuffle around framework docs

(refs #17540)

dschwen committed:
Update framework/include/markers/BoundaryMarker.h

Co-authored-by: Logan Harbour <loganhharbour@gmail.com>

dschwen committed:
Remove comments, add error check (#17683)

lindsayad committed:
Add INSFV tri and tet tests

aeslaughter committed:
Update to optional show MOOSE as dependency in SLL for applications

(refs #17540)

andrsd committed:
Do not execute tests that require SLEPc if SLEPc is not available

Refs #0

aeslaughter committed:
Remove non-tested requirements, since they are not requirements

(refs #16660)

aeslaughter committed:
Update STP for PLN-4005 v8 revisions

(refs #17540)

aeslaughter committed:
Add additional line from PLN-4005 from 9.5.1.1

(refs #17540)

dschwen committed:
Add doc page (#14974)

dschwen committed:
Implement material time step limit, address comments (#14974)

dschwen committed:
Temporarily remove creep interface (revert this later) (#14974)

yjung-anl committed:
renamed HFEM kernels and BCs

dschwen committed:
Add postprocessor value coupling (#17745)

yjung-anl committed:
added standard variable version of HFEM Diffusion kernels and BCs

dschwen committed:
Going the extra mile (#14974)

dschwen committed:
Simplify material constant handling (#14974)

dschwen committed:
Doco (#17745)

dschwen committed:
Move .f plugins into test app, fix citation, nan celent (#14974)

zachmprince committed:
Ability to specify reporter type in MultiAppCloneReporterTransfer in order to get around a processor not owning a sub-application #17751

Commit:
update to katex v0.13.5
close #17753

somu15 committed:
Renaming FluidStructureInterface

GiudGiud committed:
Add MMS testing of flat and continuously varying porosity
Re-add momentum diffusion porosity gradient term
Interpolate 1/eps term alongside the velocity (so intersititial velocity instead of superficial)
Re-gold porosity jump terms

somu15 committed:
Re-adding forcing csv

dschwen committed:
Fix tensor mechanics action material output (#17606)

somu15 committed:
Rename interface docs

dschwen committed:
Remove broken citation which we do not have enough data for (#14974)

lindsayad committed:
Add ADNodalKernel

Closes #17759

lindsayad committed:
Protect against zero size dof dependence before libMesh constraints

If we call `constrain_element_matrix` with zero size column indices then
we get errors out of BLAS. This was happening with global AD indexing
since global AD indexing in general only inserts nonzero degree of
freedom dependence.

Test to come. Necessary condition to trigger this error:
- using AD objects with global AD indexing when said objects may not
  have any nonlinear variable dependence (like `ADBodyForce`)
- Have some libmesh constraints like hanging nodes or periodic
  boundaries

Closes #17755

lindsayad committed:
Test zero-dof-dependence AD kernel with constraint

dschwen committed:
Address comments (#17745)

dschwen committed:
Fix fortran warning, clean up indentation (#17766)

gridley committed:
fix myriad typos refs #17769

GiudGiud commented on modules/stochastic_tools/include/samplers/AIS.h:
```diff
@@ -0,0 +1,80 @@
+
+protected:
+  /// Return the sample for the given row and column
+  virtual Real computeSample(dof_id_type row_index, dof_id_type col_index) override;
```
add doxygen with description of parameters
I d like to understand why col_index = 0 is special

somu15 replied:
The sampler cycles through all the parameters using col_index. For MCMC sampling, we need to propose all the next parameters at once. So, I am doing this proposal at col_index = 0 and storing these proposals. Then for col_index>0, I am simply returning the stored values. This is different from independent Monte Carlo where we can sample a parameter independently of the other parameters.

GiudGiud commented on modules/stochastic_tools/src/utils/AdaptiveMonteCarloUtils.C:
```diff
@@ -0,0 +1,42 @@
+#include "AdaptiveMonteCarloUtils.h"
+
+/* AdaptiveMonteCarloUtils contains functions that are used across the Adaptive Monte
+Carlo set of algorithms.*/
```
```suggestion
   Carlo set of algorithms.*/
```

GiudGiud commented on modules/stochastic_tools/src/utils/AdaptiveMonteCarloUtils.C:
```diff
@@ -0,0 +1,42 @@
+  }
+  for (unsigned int i = start_index; i < data.size(); ++i)
+  {
+    sq_diff1 += std::pow(((data[i])-sum1/(data.size()-start_index)), start_index);
```
```suggestion
    sq_diff1 += std::pow((data[i]-sum1/(data.size()-start_index)), start_index);
```
power of start_index? This doesnt seem right. Can you please check

somu15 replied:
Yes, this should be 2. Thanks for pointing out.

GiudGiud commented on modules/stochastic_tools/src/utils/AdaptiveMonteCarloUtils.C:
```diff
@@ -0,0 +1,42 @@
+  Real sum1 = 0.0;
+  for (unsigned int i = start_index; i < data.size(); ++i)
+  {
+    sum1 += (data[i]);
```
```suggestion
    sum1 += data[i];
```

GiudGiud reviewed (commented):
Mostly form stuff.
I ll let @zachmprince dive into the details. 

Thanks for the PR! This is pretty important framework-wide capability

loganharbour committed:
Fix poor logic in ParallelStudy work buffer resize

refs #17771

GiudGiud committed:
Add requirements, design and issues for SQA purposes to 1FV + rest FE NS tests, refs #15968

GiudGiud committed:
Add documentation for old NS FE kernels, add class descriptions to source files to show in doco
Proper formatting of tests files requirements

somu15 commented:
> Mostly form stuff.
> I ll let @zachmprince dive into the details.
> 
> Thanks for the PR! This is pretty important framework-wide capability

Thanks for the review @GiudGiud I will address your comments as soon as possible.

lindsayad committed:
Apply suggestions from code review

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

rwcarlsen committed:
allow app-custom src dependencies during compilation

This is useful for apps that may have moose-external dependencies that
they need to compile/build.  Sometimes a moose app file (e.g.
FooMaterial.C) can depend on a file generated by the build of an
external app dependency.  Without this addition - there is not way for
the app developer to tell us about this dependency - which prevents
parallel builds from working properly.

Fixes #17800.

aeslaughter committed:
Fix/test joining of urls in git submodule-hash command

(closes #17825)
(refs #17715)

aeslaughter committed:
Support toggle for including non-testable tests

(fixes #17715)

fdkong committed:
Fixed comipler issue: 'CompareTypes' is ambiguous

Closes #17833

hugary1995 committed:
ADDerivativeParsedMaterial should create material properties even if they are zero

close #15966

aeslaughter committed:
Capture CIVET results from each listed remote in documentation

(fixes #17842)

somu15 committed:
Changing adaptivemontecarloutils

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/741822/) on 6a235ab wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/17665/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/17665/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 911fe6223adc5d1a64e214b4762653eee33c380e`
